x-pack/plugins/secops/public/components/page/network/ip_overview/index.tsx

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiDescriptionList,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiText,
+  EuiToolTip,
+} from '@elastic/eui';
+import { EuiSpacer } from '@elastic/eui';
+import { FormattedRelative } from '@kbn/i18n/react';
+
+import React from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+import { ActionCreator } from 'typescript-fsa';
+
+import { IpOverviewData, IpOverviewType, Overview } from '../../../../graphql/types';
+import { networkActions, networkModel, networkSelectors, State } from '../../../../store';
+import { getEmptyTagValue } from '../../../empty_value';
+
+import {
+  autonomousSystemRenderer,
+  dateRenderer,
+  hostIdRenderer,
+  hostNameRenderer,
+  locationRenderer,
+  reputationRenderer,
+  whoisRenderer,
+} from './field_renderers';
+import { SelectType } from './select_type';
+import * as i18n from './translations';
+
+export const IpOverviewId = 'ip-overview';
+
+const SelectTypeItem = styled(EuiFlexItem)`
+  min-width: 180px;
+`;
+
+interface DescriptionList {
+  title: string;
+  description: React.ReactElement;
+}
+
+interface OwnProps {
+  ip: string;
+  data: IpOverviewData;
+  loading: boolean;
+  startDate: number;
+  endDate: number;
+  type: networkModel.NetworkType;
+}
+
+interface IpOverviewReduxProps {
+  flowType: IpOverviewType;
+}
+
+interface IpOverViewDispatchProps {
+  updateIpOverviewFlowType: ActionCreator<{
+    flowType: IpOverviewType;
+  }>;
+}
+
+type IpOverviewProps = OwnProps & IpOverviewReduxProps & IpOverViewDispatchProps;
+
+class IpOverviewComponent extends React.PureComponent<IpOverviewProps> {
+  public render() {
+    const { ip, data, loading, flowType } = this.props;
+    const typeData: Overview = data[flowType]!;
+
+    const descriptionLists: Readonly<DescriptionList[][]> = [
+      [
+        {
+          title: i18n.LOCATION,
+          description: locationRenderer(
```

Comment:
I am wondering what is the benefits of having function vs component with Props.

Reply:
Is it looks weird to put a JSX element in a config-like thing? I couldn't explain either.

Reply:
Definitely want to refactor these to be generic `pure components`. I think there's more cleanup around taking specific fields as opposed to full `ECS Types` as we talked about too. If okay, I would like to do these refactorings as part of updating the `Host Summary` UI to match the latest mockups / this `IP Overview component` in effort to support all the functionality required by the Host component as well.